### PR TITLE
holdings: add item counts

### PR DESCRIPTION
**Pull request by jma**
* Adds the number of items in the holding index.
* Changes the indexing chain to items -> holdings -> documents.
* Fixes test fixtures.
* Removes some useless index flush in the tests.

Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>

## Why are you opening this PR?

- Which task/US does it implement?
- Which issue does it fix?

## Dependencies

My PR depends on the following `rero-ils-ui`'s PR(s):

* rero/rero-ils-ui#<xx>

## How to test?

- What command should I have to run to test your PR?
- What should I test through the UI?

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

File: rero_ils/modules/holdings/listener.py [hunk 1]
```diff
@@ -31,17 +32,20 @@ def enrich_holding_data(sender, json=None, record=None, index=None,
...
+        # get the number of items for ui paging
+        item_search = ItemsSearch()[0:0].filter(
+            'term', holding__pid=record.pid)
+        # to compute the number of masked item
+        item_search.aggs.bucket('public_items', 'terms', field='_masked')
+        results = item_search.source(['organisation', 'library']).execute()
+        # number of items
+        json['items_count'] = results.hits.total.value
+        # number of masked items
+        n_item_masked = 0
```
Comment (sebastiendeleze) [inline rero_ils/modules/holdings/listener.py:46]: Better an explicit name like `number_of_masked_items`, but no matter

File: rero_ils/modules/record_extensions.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        # do nothing if already exists
+        if record.get('organisation') and record.get('library'):
+            return
+        self._add_org_and_lib(record)
+        # required for validation
+        if record.model:
+            record.model.data = record
+
+    def pre_commit(self, record):
+        """Called before a record is committed."""
```
Comment (sebastiendeleze) [inline rero_ils/modules/record_extensions.py:41]: Missing doc for argument

File: rero_ils/modules/record_extensions.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        self._add_org_and_lib(record)
+
+    def _add_org_and_lib(cls, record):
+        """Build $ref for the organisation and library of the item."""
```
Comment (sebastiendeleze) [inline rero_ils/modules/record_extensions.py:45]: Missing doc for argument

**Review by sebastiendeleze — APPROVED** [review]

File: tests/api/documents/test_documents_rest.py [hunk 4]
```diff
@@ -152,7 +152,7 @@ def test_documents_newacq_filters(app, client,
         'invenio_records_rest.doc_list',
         view='global',
         new_acquisition='{0}:{1}'.format(past, today),
-        locatiin='loc3'
```
[the remark this answers is not in the document]
Comment (jma) [reply tests/api/documents/test_documents_rest.py:155]: Despite my old age.

**Pull request merged**